src/battle_ai_main.c

```diff
@@ -548,17 +550,9 @@ static s16 AI_CheckBadMove(u8 battlerAtk, u8 battlerDef, u16 move, s16 score)
         // check off screen
         if (IsSemiInvulnerable(battlerDef, move) && moveEffect != EFFECT_SEMI_INVULNERABLE && GetWhoStrikesFirst(battlerAtk, battlerDef, TRUE) != 1)
             RETURN_SCORE_MINUS(20);    // if target off screen and we go first, don't use move
-        
-        // check if negates type
-        switch (effectiveness)
-        {
-        case AI_EFFECTIVENESS_x0:
-            RETURN_SCORE_MINUS(20);
-            break;
-        case AI_EFFECTIVENESS_x0_25:
+
+        if (!IS_MOVE_STATUS(move) && effectiveness == AI_EFFECTIVENESS_x0_25)
```

Comment:
not `<= AI_EFFECTIVENESS_x0_25`?

Reply:
No, the x0 is already handled @ line 820.
This has the purpose to make it so it tries to not use moves that are 4x not very effective on target.